mkosi/__init__.py

```diff
@@ -2425,6 +2426,35 @@ def debootstrap_knows_arg(arg: str) -> bool:
     return bytes("invalid option", "UTF-8") not in run(["debootstrap", arg], stdout=PIPE, check=False).stdout
 
 
+@contextlib.contextmanager
+def mount_apt_override_config(root: Path) -> Iterator[None]:
+    # Ensure apt doesn't see the host's config, which might include things like
+    # apt-listchanges which will break, as the content of the guest will not
+    # match.
+    with complete_step("Mounting an empty directory over apt config…", "Unmounting empty directory over apt config…"), contextlib.ExitStack() as stack:
+        stack.enter_context(mount_bind(root / "tmp", Path("/") / "etc/apt"))
+
+        yield
+
+
+@contextlib.contextmanager
+def mount_apt_local_mirror(args: MkosiArgs, root: Path) -> Iterator[None]:
+    # Ensure apt inside the image can see the local mirror outside of it
+    mirror = args.local_mirror or args.mirror
+    if mirror and mirror.startswith("file:"):
+        # Strip leading/trailing '/' as Path() does not behave well when concatenating
+        mirror_dir = mirror[5:].strip("/")
+    else:
+        yield
+        return
+
+    with complete_step("Mounting apt local mirror…", "Unmounting apt local mirror…"), contextlib.ExitStack() as stack:
+        os.makedirs(root / mirror_dir, mode=0o755, exist_ok=True)
```

Comment:
`mount_bind()` will automatically create any directories that don't exist yet so this can be dropped